### PR TITLE
Fix spdlog leakage

### LINKED ISSUE #2: spdlog leaks into librmm (?)
Not sure if this is intended --- I downloaded the `librmm` package and inspected the files locally, and this is what I saw:
截圖 2021-04-28 下午12 06 24
There are headers and also a static library for spdlog, but should these be included in the `librmm` package? I thought `spdlog` is a standalone package?
https://github.com/conda-forge/librmm-feedstock/blob/32b6251bf7baaaa908701b3ff6ac54eba070461e/recipe/meta.yaml#L27
My guess is these are due to `spdlog` being fetched by cmake instead of conda.

cc: @kkraus14 @jakirkham @rlratzel

### PR DESCRIPTION
**Pull request by leofang**
Close #2.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,9 +23,12 @@ requirements:
     - {{ compiler('cuda') }}
     - cmake >=3.18
     - make
-  run:
+  host:
     - spdlog =1.7.0
     - thrust >=1.9.0
+  run:
+    # - spdlog =1.7.0  # <--- done via run_exports
```
Comment (jakirkham) [inline recipe/meta.yaml:30]: RMM seems to pin this down to the patch release, which that export doesn’t. Not sure if that presents an issue

As RMM is header only we may get some warnings about unused dependencies from conda-build that need to be addressed somehow

**Comment by leofang** [comment]
@jakirkham Should I point `RMM_BINARY_DIR` to `$PREFIX/lib`? Otherwise rmm insists on putting some cmake files in `$PREFIX/lib64/`.

**Comment by jakirkham** [comment]
Would add `-DCMAKE_INSTALL_LIBDIR="lib"` to the `cmake` configuration in `build.sh`. This should ensure it uses `lib` instead of `lib64`

**Comment by leofang** [comment]
Oops I just pushed. Would an absolute path work? (`-DCMAKE_INSTALL_LIBDIR="$PREFIX/lib"`)

File: recipe/meta.yaml (hunk 1 above)
Comment (leofang) [reply recipe/meta.yaml:30]: I checked that on Conda-Forge the latest `spdlog` is 1.8.5, and for the 1.7 series there's only 1.7.0, so we're safe.

**Comment by leofang** [comment]
> Oops I just pushed. Would an absolute path work? (`-DCMAKE_INSTALL_LIBDIR="$PREFIX/lib"`)

Looks like it worked!

**Comment by leofang** [comment]
@kkraus14 @jakirkham @rlratzel PTAL.

btw @rlratzel a friendly reminder: You need to accept the invitations sent by GitHub so that you can join the Conda-Forge org and that we can ping you as a maintainer in a feedstock. 🙂

**Review by kkraus14 — APPROVED** [review]

**Comment by jakirkham** [comment]
> > Oops I just pushed. Would an absolute path work? (`-DCMAKE_INSTALL_LIBDIR="$PREFIX/lib"`)
> 
> Looks like it worked!

It use to matter, but it sounds like they fixed it

**Comment by rlratzel** [comment]
> btw @rlratzel a friendly reminder: You need to accept the invitations sent by GitHub so that you can join the Conda-Forge org and that we can ping you as a maintainer in a feedstock. 🙂

Sorry about that! I just accepted the invite.

**Comment by leofang** [comment]
No worries, thanks Rick!

**Pull request merged**